src/main/java/com/bookshop/service/impl/CartServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package com.bookshop.service.impl;
+
+import com.bookshop.dto.cart.CartDto;
+import com.bookshop.dto.cart.CartItemDtoResponse;
+import com.bookshop.dto.cart.CreateCartItemDto;
+import com.bookshop.dto.cart.PutCartItemDto;
+import com.bookshop.mapper.BookMapper;
+import com.bookshop.mapper.CartItemMapper;
+import com.bookshop.mapper.CartMapper;
+import com.bookshop.model.CartItem;
+import com.bookshop.model.ShoppingCart;
+import com.bookshop.model.User;
+import com.bookshop.repository.cart.CartRepository;
+import com.bookshop.repository.cart.item.CartItemRepository;
+import com.bookshop.repository.user.UserRepository;
+import com.bookshop.service.CartService;
+import java.util.NoSuchElementException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CartServiceImpl implements CartService {
+    private final CartRepository cartRepository;
+    private final CartMapper cartMapper;
+    private final UserRepository userRepository;
+    private final BookMapper bookMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public CartDto getCartInfo(Long userId) {
+        ShoppingCart cart = findCart(userId);
+        if (cart == null) {
+            User user = findUserById(userId);
+            cart = new ShoppingCart();
+            cart.setId(userId);
+            cart.setUser(user);
+            cartRepository.save(cart);
+        }
+        return cartMapper.toCartDto(cart);
+    }
+
+    @Override
+    @Transactional
+    public CartItemDtoResponse createCartItem(Long userId, CreateCartItemDto request) {
+        ShoppingCart cart = findCart(userId);
+        CartItem item = findCartItemByBookId(cart, request.getBookId());
+        if (item != null) {
+            return updateCartItem(userId, item.getId(), cartItemMapper.toPutDto(request));
+        }
+        item = new CartItem();
+        item.setShoppingCart(cart);
+        item.setBook(bookMapper.bookFromId(request.getBookId()));
+        item.setQuantity(request.getQuantity());
+        cartItemRepository.save(item);
+        return cartItemMapper.toCreateDtoResponse(item);
+    }
+
+    @Override
+    @Transactional
+    public CartItemDtoResponse updateCartItem(
+            Long userId,
+            Long cartItemId,
+            PutCartItemDto request) {
+        ShoppingCart cart = findCart(userId);
+        CartItem item = findCartItemById(cart, cartItemId);
+        item.setQuantity(request.getQuantity());
+        cartItemRepository.save(item);
+        return cartItemMapper.toCreateDtoResponse(item);
+    }
+
+    @Override
+    @Transactional
+    public void removeCartItem(Long userId, Long cartItemId) {
+        ShoppingCart cart = findCart(userId);
+        findCartItemById(cart, cartItemId);
```

Comment:
What is the purpose of calling this method without usage of value it returned? If purpose is to check is cartItem exists you should rename method